File: _includes/SampleCode/SampleCode/Samples/Graph/LikeAction.cpp
#include "pch.h"
#include "FBReturnObject.h"

using namespace Windows::Foundation::Collections;
using namespace Platform;
using namespace concurrency;
using namespace winsdkfb;
using namespace winsdkfb::Graph;

namespace SampleCode
{
    void LikeAction()
    {
        // Get active session
        FBSession^ sess = FBSession::ActiveSession;
        if (sess->LoggedIn)
        {
            // Set parameters
            PropertySet^ parameters = ref new PropertySet();
            // Set Uri to like
            parameters->Insert("object", L"https://www.microsoft.com/en-us/default.aspx");

            // Create Graph API path
            String^ graphPath = sess->User->Id + L"/og.likes";

            // Create a json class factory with a class (FBReturnObject class)
            // that can receive and parse the json response returned
            FBJsonClassFactory^ fact = ref new FBJsonClassFactory([](String^ JsonText) ->
                Object^
            {
                auto returnObject = ref new FBReturnObject();
                returnObject->Id = Windows::Data::Json::JsonObject::Parse(JsonText)->GetNamedString("id");
                return returnObject;
            });

            FBSingleValue^ sval = ref new FBSingleValue(graphPath, parameters, fact);
            create_task(sval->PostAsync()).then([=](FBResult^ result)
            {
                if (result->Succeeded)
                {
                    FBReturnObject^ response = static_cast<FBReturnObject ^>(result->Object);
                }
                else
                {
                    // Posting failed
                }
            });
        }
    }
}

File: _includes/SampleCode/SampleCodeCs/Samples/Graph/LikeAction.cs
using System;
using Windows.Foundation.Collections;
using Newtonsoft.Json;
using winsdkfb;
using winsdkfb.Graph;

namespace SampleCode
{
    public static partial class SampleFunc
    {
        static async void LikeAction()
        {
            // Get active session
            FBSession sess = FBSession.ActiveSession;
            if (sess.LoggedIn)
            {
                //Get current user
                FBUser user = sess.User;
                // Set parameters
                PropertySet parameters = new PropertySet();
                // Set Uri to like
                parameters.Add("object", "https://www.microsoft.com/en-us/default.aspx");
                // Set Graph api path
                string path = user.Id + "/og.likes";
                var factory = new FBJsonClassFactory(s =>
                {
                    return JsonConvert.DeserializeObject<FBReturnObject>(s);
                });
                var singleValue = new FBSingleValue(path, parameters, factory);
                var result = await singleValue.PostAsync();
                if (result.Succeeded)
                {
                    var response = result.Object as FBReturnObject;
                }
                else
                {
                    // Posting failed
                }
            }
        }
    }
}